### PR TITLE
MDEV-13336 Add --ignore-database option to mysqldump

### PR DESCRIPTION
**Pull request by hholzgra**
Allow to exclude certain databases from an --all-databases dump,
e.g. to be able to do

  mysqldump --all-databases --ignore-database=mysql

to dump everything but the system tables.

### CODE REVIEW

File: client/mysqldump.c [hunk 1]
```diff
@@ -899,6 +906,10 @@ get_one_option(int optid, const struct my_option *opt __attribute__((unused)),
   case (int) OPT_TABLES:
     opt_databases=0;
     break;
+  case (int) OPT_IGNORE_DATABASE:
+    if (my_hash_insert(&ignore_database, (uchar*)my_strdup(argument, MYF(0))))
```
Comment (svoj) [inline client/mysqldump.c]: There should be space after a cast: ```(uchar*) my_strdup...```

File: client/mysqldump.c [hunk 2]
```diff
@@ -1050,6 +1064,12 @@ static int get_options(int *argc, char ***argv)
             my_progname_short);
     return(EX_USAGE);
   }
+  if (ignore_database.records && !opt_alldbs) {
```
Comment (svoj) [inline client/mysqldump.c]: Coding style:
```
if ()
{
}
```

instead of

```
if () {
}
```

File: client/mysqldump.c [hunk 3]
```diff
@@ -4389,6 +4411,14 @@ static int dump_tablespaces(char* ts_where)
   DBUG_RETURN(0);
 }
 
+/* Return 1 if we should copy the database */
+
+my_bool include_database(const uchar *hash_key, size_t len)
```
Comment (svoj) [inline client/mysqldump.c]: ```static my_bool include_database...``` please.

File: client/mysqldump.c [hunk 4]
```diff
@@ -4389,6 +4411,14 @@ static int dump_tablespaces(char* ts_where)
   DBUG_RETURN(0);
 }
```
Comment (svoj) [inline client/mysqldump.c]: There should be 2 vertical spaces between functions.

File: client/mysqldump.c [hunk 5]
```diff
@@ -4430,8 +4461,9 @@ static int dump_all_databases()
           !my_strcasecmp(&my_charset_latin1, row[0], PERFORMANCE_SCHEMA_DB_NAME))
         continue;
 
-      if (dump_all_views_in_db(row[0]))
-        result=1;
+      if (include_database(row[0], strlen(row[0])))
```
Comment (svoj) [inline client/mysqldump.c]: Any special reason to do ```strlen()``` here, but not inside ```include_database()```?

File: client/mysqldump.c [hunk 6]
```diff
@@ -6005,6 +6037,7 @@ int main(int argc, char **argv)
   sf_leaking_memory=1; /* don't report memory leaks on early exits */
   compatible_mode_normal_str[0]= 0;
   default_charset= (char *)mysql_universal_client_charset;
+  bzero((char*) &ignore_database, sizeof(ignore_database));
```
Comment (svoj) [inline client/mysqldump.c]: This is not needed. Global variables are guaranteed to be set to 0 initially.

File: client/mysqldump.c [hunk 7]
```diff
@@ -213,6 +213,8 @@ TYPELIB compatible_mode_typelib= {array_elements(compatible_mode_names) - 1,
 
 HASH ignore_table;
 
+HASH ignore_database;
```
Comment (svoj) [inline client/mysqldump.c]: ```static HASH ignore_database;``` please.

**Comment by svoj** [comment]
No feedback was provided for this pull request for over a month, so it is being closed. Feel free to reopen if you are able to implement requested changes.

**Comment by svaroqui** [comment]
Would be super nice to get an option to disable binlog for restoring a slave faster

**Pull request closed without merging**